test/model/avatar_url_test.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/model/avatar_url.dart';
+
+import '../example_data.dart' as eg;
+
+void main() {
+  final user = eg.user();
+  final account = eg.account(user: user);
+  
+  double defaultSize = 30;
+  double largeSize = 120;
+
+  group('GravatarUrl', () {
+    test('gravatar url', () {
+      String url = '${GravatarUrl.origin}/avatar/1234';
+      final avatarUrl = AvatarUrl.fromUserData(
+        rawAvatarUrl: url,
+        realm: account.realmUrl,
+        userId: user.userId,
+        email: user.email,
+      );
+
+      check(avatarUrl.get(defaultSize).toString()).equals('$url?s=30.0');
```

Comment:
Hmm, this ".0" seems undesired. Is that what's seen in the zulip-mobile tests?

Reply:
OK, seems like the zulip-mobile tests don't check this question — but in that case, what's the actual behavior in zulip-mobile? Does it ever put something like "30.0" into the URL, or is it always an integer?

---

I just took a look, and it looks like all the call sites of AvatarURL.get are like this one (in `src/common/UserAvatar.js`):
```js
          uri: avatarUrl.get(PixelRatio.getPixelSizeForLayoutSize(size)).toString(),
```
in that they use `PixelRatio.getPixelSizeForLayoutSize` to get the value to pass as `sizePhysicalPx`.

And if you click through in your IDE to the definition of that method (it's part of React Native), its jsdoc says:
```js
  /**
   * Converts a layout size (dp) to pixel size (px).
   *
   * Guaranteed to return an integer number.
   */
```

So it always returns an integer, which means we always pass an integer to the `get` method, which means we always say something like `s=30` here, not `s=30.0`.

Thinking about the semantics of this `get` method and its parameter, we're talking about the size of a desired image. The image is a raster (typically a PNG, perhaps always so), which means it's represented as a grid of pixels. So the image's actual size in each dimension, as a raster, is an integer — there might be 30 rows of pixels in the grid, or 31, but there can't be 30.1 rows of pixels. So: the parameter type for `AvatarUrl.get` should be `int`, not `double`.

That distinction doesn't exist in JavaScript — there's just numbers, meaning floating-point numbers — which is why it's `number` in zulip-mobile.

Reply:
[7a0b600](https://github.com/zulip/zulip-flutter/pull/534/commits/7a0b60054e634aa5120ee821b80fe8a554ed0a77)